File: php/src/Advertisements/User/Domain/Exceptions/InvalidUserException.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\Domain\Exceptions;

use Demo\App\Common\Domain\DomainException;

final class InvalidUserException extends DomainException
{
    public const string INVALID_USER_MESSAGE = 'Invalid user';

    private function __construct(string $message)
    {
        parent::__construct($message);
    }

    public static function build(string $email): self
    {
        return new self(self::INVALID_USER_MESSAGE);
    }

    public function message(): string
    {
        return $this->getMessage();
    }
}

File: kotlin/src/main/kotlin/advertisements/user/domain/exceptions/InvalidUserException.kt
package advertisements.user.domain.exceptions

import common.domain.DomainException

class InvalidUserException private constructor(message: String) : DomainException(message) {

    companion object {
        private const val INVALID_USER_MESSAGE = "Invalid user"

        fun build(message: String): InvalidUserException {
            return InvalidUserException(INVALID_USER_MESSAGE)
        }
    }

    fun message(): String {
        return this.message ?: ""
    }
}